### PR TITLE
DEVENGAGE-1653: Added encoding for url path variables in go sdk

### PR DESCRIPTION
**Pull request by dginty4**

### CODE REVIEW

File: resources/sdk/purecloudgo/templates/apiclient.mustache [hunk 1]
```diff
@@ -206,23 +206,6 @@ func (c *APIClient) CallAPI(path string, method string,
...
-		urlString = urlString[:len(urlString)-1]
-	}
-
-	u, err := url.Parse(urlString)
-	if err != nil {
-		return nil, err
-	}
-	return u, nil
-}
-
```
Comment (dginty4) [inline resources/sdk/purecloudgo/templates/apiclient.mustache:225]: I noticed this function wasn't being used anymore so i just removed it

**Review by HemanthDogiparthi12 — APPROVED** [review]

File: resources/sdk/purecloudgo/templates/apiclient.mustache (hunk 1 above)
Comment (charliecon) [reply resources/sdk/purecloudgo/templates/apiclient.mustache:225]: @dginty4 Have you run a test build on jenkins? I'd do that just to be sure removing this function didn't cause any issues

Comment (dginty4) [reply resources/sdk/purecloudgo/templates/apiclient.mustache:225]: Good idea, thanks Charlie

Comment (dginty4) [reply resources/sdk/purecloudgo/templates/apiclient.mustache:225]: @charliecon it built on Jenkins

**Pull request merged**